### PR TITLE
Remove beacon rpc service

### PR DESCRIPTION
**Pull request by terencechain**
There is `beacon chain service` which implements with ethereumAPI, then there is `beacon service` which implements `WaitForChainStart`. 
This PR removes `beacon service` and moves `WaitForChainStart` and tests to `validator services`

### CODE REVIEW

**Comment by nisdas** [comment]
@terencechain this looks fine overall, have you tried this at runtime to confirm that it works as expected ?

**Comment by terencechain** [comment]
> @terencechain this looks fine overall, have you tried this at runtime to confirm that it works as expected ?

Yep. Single node and multi nodes with eth1 mock rpc all working

File: beacon-chain/rpc/proposer_server.go [hunk 1]
```diff
@@ -306,3 +306,15 @@ func (ps *ProposerServer) defaultEth1DataResponse(ctx context.Context, currentHe
 		DepositCount: depositsTillHeight,
 	}, nil
 }
+
+func constructMerkleProof(trie *trieutil.MerkleTrie, index int, deposit *ethpb.Deposit) (*ethpb.Deposit, error) {
```
Comment (nisdas) [inline beacon-chain/rpc/proposer_server.go]: also is this used anywhere ?

Comment (terencechain) [reply beacon-chain/rpc/proposer_server.go]: Yep used in proposer server

**Pull request merged**